src/prism.c

```diff
@@ -15383,6 +15402,10 @@ parse_regular_expression_named_captures(pm_parser_t *parser, const pm_string_t *
             const uint8_t *source = pm_string_source(name);
             size_t length = pm_string_length(name);
 
+            if (!name_is_identifier(parser, source, length)) {
```

Comment:
I think this is good, but in this case it's possible to have a `MatchWriteNode` when there aren't actually any matches. In this case I think we shouldn't replace the call node with a match write. Instead, we could lazily create the `match` node once we are sure we have a named capture group that is valid, and then only set `result = match` if one was created.